lib/Disco.php

```diff
@@ -941,28 +941,32 @@ private function fillSpName($t)
                     }
                 }
             } else {
-                $entityId = $this->originalsp['entityid'];
-                $entityIdAttr = $this->wayfConfiguration->getString(self::ENTITY_ID_ATTR, null);
-                if (null === $entityIdAttr) {
-                    $facility = $this->adapter->getFacilityByEntityId($entityId);
-                } else {
-                    $facility = $this->adapter->getFacilityByEntityId($entityId, $entityIdAttr);
-                }
+                if (!empty($this->originalsp['entityid'])) {
+                    $entityId = $this->originalsp['entityid'];
+                    $entityIdAttr = $this->wayfConfiguration->getString(self::ENTITY_ID_ATTR, null);
+                    if (null === $entityIdAttr) {
+                        $facility = $this->adapter->getFacilityByEntityId($entityId);
+                    } else {
+                        $facility = $this->adapter->getFacilityByEntityId($entityId, $entityIdAttr);
+                    }
+
+                    if (null === $facility) {
```

Comment:
Some parts of the else branch are the same as in the if.

So I think it could be rewritten. @dBucik ??

Reply:
Refactored, it should be less repetitive now